In [1]:
# Example: Using Dask to process a large CSV file in parallel
# Install Dask if not already installed: !pip install dask[complete]
import dask.dataframe as dd

csv_path = r"C:\Users\ayo\Downloads\internship_dataset\OECD.DCD.FSD_DSD_CRS@DF_CRS_1_4_agri.csv"

# Read the CSV file as a Dask DataFrame (Dask will use all available CPU cores)
ddf = dd.read_csv(csv_path, blocksize="64MB", assume_missing=True)

# Example: Get the shape (rows, columns)
row_count = ddf.shape[0].compute()
col_count = ddf.shape[1]
print(f"Dask DataFrame shape: {row_count} rows, {col_count} columns")

# Example: Find columns with more than 90% missing values
total_rows = row_count
missing_counts = ddf.isnull().sum().compute()
mostly_empty_cols = missing_counts[missing_counts / total_rows > 0.9].index.tolist()
print(f"Columns with more than 90% missing values: {mostly_empty_cols}")

C:\Users\ayo\AppData\Roaming\Python\Python313\site-packages\dask\dataframe\io\csv.py:77: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = reader(bio, **kwargs)
C:\Users\ayo\AppData\Roaming\Python\Python313\site-packages\dask\dataframe\io\csv.py:77: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = reader(bio, **kwargs)
C:\Users\ayo\AppData\Roaming\Python\Python313\site-packages\dask\dataframe\io\csv.py:77: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = reader(bio, **kwargs)
C:\Users\ayo\AppData\Roaming\Python\Python313\site-packages\dask\dataframe\io\csv.py:77: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = reader(bio, **kwargs)
C:\Users\ayo\AppData\Roaming\Python\Python313\site-packages\dask\dataframe\io\csv.py:77: DtypeWarning: Columns (11) have mix

Dask DataFrame shape: 78100206 rows, 103 columns


C:\Users\ayo\AppData\Roaming\Python\Python313\site-packages\dask\dataframe\io\csv.py:77: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = reader(bio, **kwargs)
C:\Users\ayo\AppData\Roaming\Python\Python313\site-packages\dask\dataframe\io\csv.py:77: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = reader(bio, **kwargs)
C:\Users\ayo\AppData\Roaming\Python\Python313\site-packages\dask\dataframe\io\csv.py:77: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = reader(bio, **kwargs)
C:\Users\ayo\AppData\Roaming\Python\Python313\site-packages\dask\dataframe\io\csv.py:77: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = reader(bio, **kwargs)
C:\Users\ayo\AppData\Roaming\Python\Python313\site-packages\dask\dataframe\io\csv.py:77: DtypeWarning: Columns (11) have mix

Columns with more than 90% missing values: ['Microdata ID', 'Observation value', 'Base period', 'DONOR_AGENCY', 'Donor agency', 'OECD_ID', 'OECD ID', 'DONOR_PROJECT_ID', 'Donor project ID', 'REGION', 'Region', 'INCOME_GROUP', 'Income group', 'CHANNELDELIVERY_CODE', 'Channel of delivery code', 'CHANNELDELIVERY_NAME', 'Channel of delivery name', 'CATEGORY_CODE', 'Category code', 'CATEGORY_NAME', 'Category name', 'FINANCETYPE_CODE', 'FinanceType code', 'FINANCETYPE_NAME', 'FinanceType name', 'BIMULTI_CODE', 'BiMulti code', 'SHORT_DESCRIPTION', 'Short description', 'PROJECT_TITLE', 'Project title', 'LONG_DESCRIPTION', 'Long description', 'SDG_FOCUS', 'SDG focus', 'GENDER_EQUALITY', 'Gender equality', 'ENVIRONMENT', 'Environment', 'GOVERNANCE', 'Governance', 'RMNCH', 'RMNCH.1', 'DRR', 'DRR.1', 'NUTRITION', 'Nutrition', 'DISABILITY', 'Disability', 'BIODIVERSITY', 'Biodiversity', 'CLIMATE_MITIGATION', 'Climate mitigation', 'CLIMATE_ADAPTATION', 'Climate adaptation', 'DESERTIFICATION', 'Desert

Dask DataFrame shape: 78100206 rows, 103 columns
Columns with more than 90% missing values: ['Microdata ID', 'Observation value', 'Base period', 'DONOR_AGENCY', 'Donor agency', 'OECD_ID', 'OECD ID', 'DONOR_PROJECT_ID', 'Donor project ID', 'REGION', 'Region', 'INCOME_GROUP', 'Income group', 'CHANNELDELIVERY_CODE', 'Channel of delivery code', 'CHANNELDELIVERY_NAME', 'Channel of delivery name', 'CATEGORY_CODE', 'Category code', 'CATEGORY_NAME', 'Category name', 'FINANCETYPE_CODE', 'FinanceType code', 'FINANCETYPE_NAME', 'FinanceType name', 'BIMULTI_CODE', 'BiMulti code', 'SHORT_DESCRIPTION', 'Short description', 'PROJECT_TITLE', 'Project title', 'LONG_DESCRIPTION', 'Long description', 'SDG_FOCUS', 'SDG focus', 'GENDER_EQUALITY', 'Gender equality', 'ENVIRONMENT', 'Environment', 'GOVERNANCE', 'Governance', 'RMNCH', 'RMNCH.1', 'DRR', 'DRR.1', 'NUTRITION', 'Nutrition', 'DISABILITY', 'Disability', 'BIODIVERSITY', 'Biodiversity', 'CLIMATE_MITIGATION', 'Climate mitigation', 'CLIMATE_ADAPTATION', 'Climate adaptation', 'DESERTIFICATION', 'Desertification', 'AMOUNT_TIED', 'Amount tied', 'AMOUNT_UNTIED', 'Amount untied', 'AMOUNT_PARTIALLY_UNTIED', 'Amount partially untied', 'NATURE_OF_SUBMISSION', 'Nature of submission', 'PSIFLAG_CODE', 'PSI flag', 'PSIFLAG_NAME', 'PSI Flag name']

In [2]:
len(mostly_empty_cols)  # Count of columns with more than 90% missing values

69

In [3]:
103-len(mostly_empty_cols)  

34